crates/ruff/src/rules/flake8_todos/rules.rs

```diff
@@ -289,44 +288,34 @@ static ISSUE_LINK_REGEX_SET: Lazy<RegexSet> = Lazy::new(|| {
     .unwrap()
 });
 
-pub(crate) fn todos(tokens: &[LexResult], settings: &Settings) -> Vec<Diagnostic> {
+pub(crate) fn todos(indexer: &Indexer, locator: &Locator, settings: &Settings) -> Vec<Diagnostic> {
     let mut diagnostics: Vec<Diagnostic> = vec![];
 
-    let mut iter = tokens.iter().flatten().multipeek();
-    while let Some((token, token_range)) = iter.next() {
-        let Tok::Comment(comment) = token else {
-            continue;
-        };
+    let mut iter = indexer.comment_ranges().iter().multipeek();
+    while let Some(comment_range) = iter.next() {
+        let comment = locator.slice(*comment_range);
 
         // Check that the comment is a TODO (properly formed or not).
-        let Some(tag) = detect_tag(comment, token_range.start()) else {
+        let Some(tag) = detect_tag(comment, comment_range.start()) else {
             continue;
         };
 
         tag_errors(&tag, &mut diagnostics, settings);
-        static_errors(&mut diagnostics, comment, *token_range, &tag);
+        static_errors(&mut diagnostics, comment, *comment_range, &tag);
 
         // TD003
         let mut has_issue_link = false;
-        while let Some((token, token_range)) = iter.peek() {
-            match token {
-                Tok::Comment(comment) => {
-                    if detect_tag(comment, token_range.start()).is_some() {
-                        break;
-                    }
-                    if ISSUE_LINK_REGEX_SET.is_match(comment) {
-                        has_issue_link = true;
-                        break;
-                    }
-                }
-                Tok::Newline | Tok::NonLogicalNewline => {
-                    continue;
-                }
-                _ => {
-                    break;
-                }
```

Comment:
I think we still need some handling that tests whether the two comments are next to each other (or have only trivia in between). We can do this by testing whether the range between `range.end()` and `next_comment_range.start()` only contains whitespace.

Reply:
RE this comment and below

I just wanted to change the arguments to the respective checkers and get the PR up - still need to remove tokens from use. 

Sorry though, what's trivia?

Reply:
> Sorry though, what's trivia?

Sorry for the late reply. Trivia are tokens that have no semantic meaning, e.g. whitespace, newlines (Python is a bit awkward because some newlines have semantic meaning), comments etc. I think in this context it is sufficient to test if there's any content between the two comments.